pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -170,9 +170,11 @@ public SuccessResponse addSchema(
   @ApiResponses(value = {@ApiResponse(code = 200, message = "Successfully validated schema"), @ApiResponse(code = 400, message = "Missing or invalid request body"), @ApiResponse(code = 500, message = "Internal error")})
   public String validateSchema(FormDataMultiPart multiPart) {
     Schema schema = getSchemaFromMultiPart(multiPart);
-    if (!SchemaUtils.validate(schema, LOGGER)) {
-      throw new ControllerApplicationException(LOGGER, "Invalid schema. Check controller logs",
-          Response.Status.BAD_REQUEST);
+    try {
+      SchemaUtils.validate(schema);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, String.format("Invalid schema:%s.", schema.getSchemaName()),
```

Comment:
(nit) Space after ':' and no '.' after the schema name for readability. Same for other exceptions
```suggestion
      throw new ControllerApplicationException(LOGGER, "Invalid schema: " + schema.getSchemaName()),
```